tests/Unit/ProtoTests/Basic/out/src/Client/BasicClient.php

```diff
@@ -208,4 +211,28 @@ public function methodWithArgs(RequestWithArgs $request, array $callOptions = []
     {
         return $this->startApiCall('MethodWithArgs', $request, $callOptions)->wait();
     }
+
+    /** Configure the gapic configuration to use a service emulator. */
+    private function setEmulatorConfig()
+    {
+        $emulatorHost = getenv('BASIC_EMULATOR_HOST');
+        if (!empty($emulatorHost)) {
+            if (parse_url($emulatorHost, PHP_URL_SCHEME) === $scheme) {
+                $search = $scheme . '://';
+                $emulatorHost = str_replace($search, '', $emulatorHost);
+            }
+
+            return [
+                'apiEndpoint' => $emulatorHost,
```

Comment:
I think we are planning to move away from `apiEndpoint`.
I am not sure whether we should have `serviceAddress` or some other field instead.

Reply:
I see `apiEndpoint` used in the Gapic generated Admin clients here:             'apiEndpoint' => self::SERVICE_ADDRESS . ':' . self::DEFAULT_SERVICE_PORT,

Reply:
Using `apiEndpoint` is the correct option. The deprecated field is `serviceAddress`

Reply:
Thank you Brent for the clarification